[BUG]: Performing a non-debug build yields several compiler warnings not seen during debug builds

### Version

24.10

### Which installation method(s) does this occur on?

Source

### Describe the bug.

```
/home/dagardner/work/conda/envs/morpheus/envs/morpheus-2410reltest/include/boost/fiber/buffered_channel.hpp:71:18: warning: argument 1 value '18446744073709551615' exceeds maximum object size 9223372036854775807 [-Walloc-size-larger-than=]
   71 |         slots_ = new slot_type[capacity_];
      |                  ^
/home/dagardner/work/conda/envs/morpheus/envs/morpheus-2410reltest/x86_64-conda-linux-gnu/include/c++/12.1.0/new: In member function 'get_source':
/home/dagardner/work/conda/envs/morpheus/envs/morpheus-2410reltest/x86_64-conda-linux-gnu/include/c++/12.1.0/new:128:26: note: in a call to allocation function 'operator new []' declared here
  128 | _GLIBCXX_NODISCARD void* operator new[](std::size_t) _GLIBCXX_THROW (std::bad_alloc)
```

### Minimum reproducible example

```bash
./scripts/compile.sh
```

### Relevant log output

<details><summary>Click here to see error details</summary><pre>

 [Paste the error here, it will be hidden by default]

</pre></details>


### Full env printout

<details><summary>Click here to see environment details</summary><pre>

 [Paste the results of print_env.sh here, it will be hidden by default]

</pre></details>


### Other/Misc.

_No response_

### Code of Conduct

- [x] I agree to follow Morpheus' Code of Conduct
- [x] I have searched the [open bugs](https://github.com/nv-morpheus/Morpheus/issues?q=is%3Aopen+is%3Aissue+label%3Abug) and have found no duplicates for this bug report

--- a/examples/developer_guide/3_simple_cpp_stage/CMakeLists.txt
+++ b/examples/developer_guide/3_simple_cpp_stage/CMakeLists.txt
@@ -38,6 +38,10 @@ set(CMAKE_INSTALL_RPATH "$ORIGIN")
 
 # Set the option prefix to match the outer project before including. Must be before find_package(morpheus)
 set(OPTION_PREFIX "MORPHEUS")
+
+# Set the policy to allow for CMP0144, avoids warning about MORPHEUS_ROOT being set
+cmake_policy(SET CMP0144 NEW)
+
 find_package(morpheus REQUIRED)
 
 morpheus_utils_initialize_cpm(MORPHEUS_CACHE_DIR)

--- a/examples/developer_guide/4_rabbitmq_cpp_stage/CMakeLists.txt
+++ b/examples/developer_guide/4_rabbitmq_cpp_stage/CMakeLists.txt
@@ -38,6 +38,10 @@ set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
 
 # Set the option prefix to match the outer project before including. Must be before find_package(morpheus)
 set(OPTION_PREFIX "MORPHEUS")
+
+# Set the policy to allow for CMP0144, avoids warning about MORPHEUS_ROOT being set
+cmake_policy(SET CMP0144 NEW)
+
 find_package(morpheus REQUIRED)
 
 morpheus_utils_initialize_cpm(MORPHEUS_CACHE_DIR)

--- a/python/morpheus/morpheus/_lib/cmake/libmorpheus.cmake
+++ b/python/morpheus/morpheus/_lib/cmake/libmorpheus.cmake
@@ -112,6 +112,9 @@ add_dependencies(morpheus ${cudf_helpers_target})
 # In debug mode, dont allow missing symbols
 target_link_options(morpheus PUBLIC "$<$<CONFIG:Debug>:-Wl,--no-allow-shlib-undefined>")
 
+# Avoid warning from the lto-wrapper about serial compilation
+target_link_options(morpheus PUBLIC "-flto=auto")
+
 # Generates an include file for specifying external linkage since everything is hidden by default
 generate_export_header(morpheus
   NO_EXPORT_MACRO_NAME

--- a/python/morpheus_llm/morpheus_llm/_lib/cmake/libmorpheus_llm.cmake
+++ b/python/morpheus_llm/morpheus_llm/_lib/cmake/libmorpheus_llm.cmake
@@ -60,6 +60,9 @@ target_include_directories(morpheus_llm
 # In debug mode, dont allow missing symbols
 target_link_options(morpheus_llm PUBLIC "$<$<CONFIG:Debug>:-Wl,--no-allow-shlib-undefined>")
 
+# Avoid warning from the lto-wrapper about serial compilation
+target_link_options(morpheus_llm PUBLIC "-flto=auto")
+
 # Ideally, we dont use glob here. But there is no good way to guarantee you dont miss anything like *.cpp
 file(GLOB_RECURSE morpheus_llm_public_headers
   LIST_DIRECTORIES FALSE

